### ISSUE TITLE
Can't suppress warning for a project containing no code files

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn#51868]
The `Package` project for analyzers from the default template is by-design not containing any code files, and it violates CA1014. Having `dotnet_diagnostic.CA1014.severity = none` makes no effect, whether using .editorconfig or .globalconfig.

I expect this could be by-design for .editorconfig (they require `[*.cs]` section, and the project doesn't contain any code files). However, for globalconfig I expect this to work.

**Comment by jmarolf** [comment]
This is happening because the analyzer diagnostic is reported [with no location](https://github.com/dotnet/roslyn-analyzers/blob/a0522afd52b25ad211298e308fd5bbc7734865e0/src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/MarkAssembliesWithAttributesDiagnosticAnalyzer.cs#L95-L97). The engine cannot configure this because it has no location to check against. 

I assume the fix for this would be to allow globalconfig files to suppress location-less diagnostics. @mavasani has that ever been possible before?

**Comment by mavasani** [comment]
Yes, this should work for globalconfig files. Seems to be a dupe of https://github.com/dotnet/roslyn/issues/37876#issuecomment-686511276. @Youssef1313 Can you provide a snippet of your globalconfig file?

**Comment by Youssef1313** [comment]
@jmarolf I can correctly suppress the no-location warning if there is any code file in the project. This is specific to the combination of no-location **and also** having no code files.

**Comment by Youssef1313** [comment]
> Yes, this should work for globalconfig files. Seems to be a dupe of [#37876 (comment)](https://github.com/dotnet/roslyn/issues/37876#issuecomment-686511276). @Youssef1313 Can you provide a snippet of your globalconfig file?

```
is_global = true

dotnet_diagnostic.CA1014.severity = none
```

To reproduce, there shouldn't be **any** code files in the project. Just `csproj` and `.globalconfig` files.

**Comment by mavasani** [comment]
> To reproduce, there shouldn't be any code files in the project.

Oh okay. In that case, I am not sure this will meet the bar to fix the issue.
